lib/go-tc/deliveryservice_requests.go

```diff
@@ -20,15 +20,264 @@ import (
+		{{if .OriginHeaders}}<h3>Header(s) to Access Origin</h3>
+		<ul>{{range .OriginHeaders}}
+			<li>{{.}}</li>{{end}}
+		</ul>{{end}}
+		<h3>Other Origin Security</h3>
+		<p>{{if .OtherOriginSecurity}}{{.OtherOriginSecurity}}{{else}}None{{end}}</p>
+	</details>
+</section>
+<section>
+	<details>
+		<summary><h2>Core Features</h2></summary>
+		<h3>Query String Handling</h3>
+		<p>{{.QueryStringHandling}}</p>
+		<h3>Range Request Handling</h3>
+		<p>{{.RangeRequestHandling}}</p>
+		<h3>Signed URLs / URL Tokenization</h3>
+		<p>{{.HasSignedURLs}}</p>
+		<h3>Negative Caching Customization</h3>
+		<p>{{.HasNegativeCachingCustomization}}</p>
+		{{if or .HasNegativeCachingCustomization .NegativeCachingCustomizationNote }}<aside>
+			<p>{{.NegativeCachingCustomizationNote}}</p>
+		</aside>{{else if .HasNegativeCachingCustomization}}<aside>
+			<p><b>No instructions given!</b></p>
+		</aside>{{end}}
+		{{if .ServiceAliases}}<h3>Service Aliases</h3>
+		<ul>{{range .ServiceAliases}}
+			<li>{{.}}</li>{{end}}
+		</ul>{{end}}
+	</details>
+</section>
+{{if or .RateLimitingGBPS .RateLimitingTPS .OverflowService}}<section>
+	<details>
+		<summary><h2>Service Limits</h2></summary>
+		{{if .RateLimitingGBPS}}<h3>Bandwidth Limit</h3>
+		<p>{{.RateLimitingGBPS}}GBps</p>{{end}}
+		{{if .RateLimitingTPS}}<h3>Transactions per Second Limit</h3>
+		<p>{{.RateLimitingTPS}}Tps</p>{{end}}
+		{{if .OverflowService}}<h3>Overflow Service</h3>
+		<p>{{.OverflowService}}</p>{{end}}
+	</details>
+</section>{{end}}
+{{if or .HeaderRewriteEdge .HeaderRewriteMid .HeaderRewriteRedirectRouter}}<section>
+	<details>
+		<summary><h2>Header Customization</h2></summary>
+		{{if .HeaderRewriteEdge}}<h3>Header Rewrite - Edge Tier</h3>
+		<pre>{{.HeaderRewriteEdge}}</pre>{{end}}
+		{{if .HeaderRewriteMid}}<h3>Header Rewrite - Mid Tier</h3>
+		<pre>{{.HeaderRewriteMid}}</pre>{{end}}
+		{{if .HeaderRewriteRedirectRouter}}<h3>Header Rewrite - Router</h3>
+		<pre>{{.HeaderRewriteRedirectRouter}}</pre>{{end}}
+	</details>
+</section>{{end}}
+{{if .Notes}}<section>
+	<details>
+		<summary><h2>Additional Notes</h2></summary>
+		<p>{{.Notes}}</p>
+	</details>
+</section>{{end}}
+</body>
+</html>
+`))
+
 // IDNoMod type is used to suppress JSON unmarshalling
 type IDNoMod int
 
+// DeliveryServiceRequestRequest is a literal request to make a Delivery Service.
+type DeliveryServiceRequestRequest struct {
+	// EmailTo is the email address that is ultimately the destination of a formatted DeliveryServiceRequestRequest.
+	EmailTo string `json:"emailTo"`
+	// Details holds the actual request in a data structure.
+	Details DeliveryServiceRequestDetails `json:"details"`
+}
+
+// DeliveryServiceRequestDetails
+type DeliveryServiceRequestDetails struct {
+	// ContentType is the type of content to be delivered, e.g. "static", "VOD" etc.
+	ContentType string `json:"contentType"`
+	// Customer is the requesting customer - typically this is a Tenant.
+	Customer string `json:"customer"`
```

Comment:
Some are optional and some aren't - the pointer type allows them to be null/missing

Reply:
I thought we handled that other places by just making the whole struct pointers. Im indifferent just asking for consistency reasons

Reply:
It just automatically coalesces `null` and `undefined` to empty strings for me so I don't need to do `nil` checks. 

Since the template is just an email, I figured people ought to be free to exclude information if they don't know. The information is processed by a human, not a machine, after all.

Reply:
I can change it to all pointers and use ozzo to check the fields - I don't really have a problem with that, just something I didn't bother doing really.

Reply:
Nah no need I won’t block it.